petab_select/candidate_space.py

```diff
@@ -529,6 +602,628 @@ def iterate():
+            self.models = self.inner_candidate_space.models
+            self.distances = self.inner_candidate_space.distances
+            self.exclusions = self.inner_candidate_space.exclusions
+
+            return return_value
+
+        return True
+
+    def _consider_method(self, model) -> bool:
+        """See `CandidateSpace._consider_method`."""
+        return self.inner_candidate_space._consider_method(model)
+
+    def reset_accepted(self) -> None:
+        """Changing the reset_accepted to reset the
+        inner_candidate_space as well."""
+        self.models = []
+        self.distances = []
+        self.inner_candidate_space.reset_accepted()
+
+    def set_predecessor_model(
+        self, predecessor_model: Union[Model, str, None]
+    ):
+        """Setting the predecessor model for the
+        inner_candidate_space as well."""
+        super().set_predecessor_model(predecessor_model=predecessor_model)
+        self.inner_candidate_space.set_predecessor_model(
+            predecessor_model=predecessor_model
+        )
+
+    def set_exclusions(self, exclusions: Union[List[str], None]):
+        """Setting the exclusions for the
+        inner_candidate_space as well."""
+        self.exclusions = exclusions
+        self.inner_candidate_space.exclusions = exclusions
+        if self.exclusions is None:
+            self.exclusions = []
+            self.inner_candidate_space.exclusions = []
+
+    def set_limit(self, limit: TYPE_LIMIT = None):
+        """Setting the limit for the
+        inner_candidate_space as well."""
+        if limit is not None:
+            self.limit.set_limit(limit)
+            self.inner_candidate_space.limit.set_limit(limit)
+
+    def is_plausible(self, model: Model) -> bool:
+        if self.check_critical(model):
+            return self.inner_candidate_space.is_plausible(model)
+        return False
+
+    def check_swap(self, model: Model) -> bool:
+        """Check if parameters that are swapped are contained in the
+        same swap parameter set."""
+        if self.method != Method.LATERAL:
+            return True
+
+        predecessor_estimated_parameters_ids = set(
+            self.predecessor_model.get_estimated_parameter_ids_all()
+        )
+        estimated_parameters_ids = set(model.get_estimated_parameter_ids_all())
+
+        swapped_parameters_ids = estimated_parameters_ids.symmetric_difference(
+            predecessor_estimated_parameters_ids
+        )
+
+        for swap_set in self.swap_parameter_sets:
+            if swapped_parameters_ids.issubset(set(swap_set)):
+                return True
+        return False
+
+    def check_critical(self, model: Model) -> bool:
+        """Check if the model contains all necessary critical parameters"""
+
+        estimated_parameters_ids = set(model.get_estimated_parameter_ids_all())
+        for critical_set in self.critical_parameter_sets:
+            if not estimated_parameters_ids.intersection(set(critical_set)):
```

Comment:
Do type conversion (`set(critical_set)`) once in constructor

Reply:
Yeah that makes more sense, to make critical_sets already actual sets. For me for some reason they were lists 😅

Reply:
Makes sense to have list of sets, or set of sets :+1:

Also fine to leave as is.